### ISSUE TITLE
Support marking database notification as read from action

### ISSUE THREAD

**Issue by molcsab** [laravel-filament/filament#5491]
### Package

filament/filament

### Package Version

v2.16.62

### Laravel Version

v.9.48.0

### Livewire Version

v2.11.0

### PHP Version

PHP 8.2

### Problem description

When I click on an action button in a database notification, the notification is still remains marked as unread.

Submitted based on a Discord discussion with Dan

### Expected behavior

It would be useful to have an option for marking the notification as read from the action and another one for removing it from the database.

### Steps to reproduce

Any database notification.

### Reproduction repository

https://github.com/filamentphp/demo

**Comment by zepfietje** [comment]
Removing a notification from the database on action click is already possible by calling `->close()` on the notification action. Could you confirm that works for you, @molcsab?

Similar API could be implemented for marking as read. 👍

**Comment by danharrin** [comment]
@zepfietje I think the main use case for this is an action (with a URL or something) and when you click on the URL, the notification is marked as read instantly before the user is redirected

**Comment by molcsab** [comment]
It doesn't work for me @zepfietje .

I use it like this:

```php
Action::make('view')
      ->button()
      ->close()
      ->label('View')
      ->url(ActivityResource::getUrl('view', ['record' => $this->record]))
   ])
 ```
 
 If I omit the button() option, then it removes the notification indeed.

**Comment by zepfietje** [comment]
> @zepfietje I think the main use case for this is an action (with a URL or something) and when you click on the URL, the notification is marked as read instantly before the user is redirected

Yeah makes sense. Marking as read from action still needs to be implemented.

**Comment by zepfietje** [comment]
> It doesn't work for me @zepfietje .
> 
> I use it like this:
> 
> ```
> Action::make('view')
>       ->button()
>       ->close()
>       ->label('View')
>       ->url(ActivityResource::getUrl('view', ['record' => $this->record]))
>    ])
> ```
> 
> If I omit the button() option, then it removes the notification indeed.

Hmm, might it be that Alpine `x-on:click` and `href` don't work well together?
The close method is returned from `getAlpineMountAction()`.

```
:x-on:click="$action->getAlpineMountAction()"
:wire:click="$action->getLivewireMountAction()"
:href="$isDisabled ? null : $url"
```

**Comment by danharrin** [comment]
Implemented by #6570 in v3
